third_party/java/arcs/build_defs/internal/kotlin.bzl

```diff
@@ -682,9 +687,9 @@ def arcs_kt_schema(
     """
     supported = ["jvm", "wasm"]
 
-    # TODO(#5018)
-    if "jvm" not in platforms:
-        platforms.append("jvm")
```

Comment:
@alxrsngrtn  I had to remove this to allow the build to succeed: otherwise, the wasm build was failing because when `jvm` is present in the platforms, then the JVM dep is added, which doesn't exist for wasm.

Is #5018 fixed? It appears that the build works correctly with this removed, but even though it's working now, I have another question:

We specify `platforms` as a list, and so it's possible to invoke the rule with both platforms, right? But this means that the jvm-only dep will be included in the deps for wasm as well, so such an invocation would no longer work.

It seems that the way to resolve this would be to allow a way to provide dep sets to `arcs_kt_library` that only apply to a given platform, but maybe there's another approach, as well?

Reply:
No one has looked into the wasm platform issue. It may have been fixed amid refactors and we didn't notice. I'm confident that GCB or TAP will catch the error if removing this causes a problem. Thanks for testing this out -- once this lands, feel free to close the bug. 

Yes, it's possible this rule will generate code for both platforms at once. That's a good catch, and you're fix sounds like it will work.

Another way to implement this is to make the java dep "multiplatform" with aliasing, even if it's an alias to nothing. In the `arcs_kt_libary`, if the platform is `wasm`, we [add `-wasm` to the dep](https://github.com/PolymerLabs/arcs/blob/750d7edf6ed4923564bc30a42883858d5cd670ee/third_party/java/arcs/build_defs/internal/kotlin.bzl#L262). Thus, a simple way you can make this work would be to create an dummy `jsr250_annotations-wasm` alias. The alias could be the same for `-js`, too.

The alias can be some sort of null dependency (not sure how this would be implemented) or an path to wherever the `@Generated` wasm annotation lives.

Reply:
I have a vague recollection: I think removing default jvm platform will break IDE integration (we'll see red squiggles again when editing Wasm).

Reply:
Ok, it sounds like i'd better not remove this for now, then.

I was wondering about making annotations dep "multi-platform", but then didn't suggest it. From what you've described, that seems like the right solution here (rather than trying to refactor the rules to accept platform-specific depsets). 

I'll look into adding a jsr250_annotations-wasm alias that points to a custom-created `@Generated` annotation.

Reply:
I appreciate all this bazel mucking around you're doing for an otherwise simple codegen change :)

Reply:
yeah, no problem... I'm less familiar with best practices inside of some of these files, your advice is really helpful.

I'm done making modifications for now. Any other thoughts on things that should change?

Reply:
The only thing that I can think of is: It would be nice to have an alias for `js`, so the rules don't break later on. I'm happy with filing a todo (can link to a GH issue) in the alias file for us to figure it out later.

Reply:
I could copy the Generated.kt file into the js SDK as well, and use a similar approach that I did with the WASM alias (that is, referencing the jsr250 dep simply ensures that the entire WASM sdk dep is included, and the wasm SDK dep includes the needed anotation type).